WordPress/gutenberg
# Global styles: output :root selector for CSS custom properties

## What? And why?!

This PR maybe resolves https://github.com/WordPress/gutenberg/issues/35840

What's happening is that I'm changing`body` to `:root` as the preset block selector to make it consistent with other CSS custom property blocks, [example](https://github.com/WordPress/gutenberg/blob/5424ec0b063bd541b3a2c5602e08721dae6db310/packages/block-library/src/common.scss).

<img width="412" alt="Screenshot 2023-04-05 at 1 06 12 pm" src="https://user-images.githubusercontent.com/6458278/229973547-32bd5855-03d9-42cf-8ec2-048cab6b660f.png">

## Alternative approaches

> [!NOTE]
> For the purposes of backwards compat, another idea is to create a theme.json settings entry to allow overwriting the CSS selector for CSS custom properties, e.g., `"settings.cssPropertiesSelector"` or something.



### TODO

- [ ] Look at root global padding CSS custom properties (`--wp--root--padding-*`) and backwards compat in another PR

## Testing Instructions

I'm using empty theme to test but you can use any theme! 

I've tested using 2022 and also 2021, 2020, 2019 (classic themes)

Note: for classic themes, only the preset CSS rule block (the `:root { --wp--preset--*: ... }` is output.

Check that presets work and there are no visual regressions.

Add a few custom color palette presets and check the frontend. They should be all sitting comfortably under `:root`.
